src/map/packets/char_recast.cpp

```diff
@@ -45,7 +45,12 @@ CCharRecastPacket::CCharRecastPacket(CCharEntity* PChar)
     {
         uint32 recasttime = (recast.RecastTime == 0 ? 0 : ((recast.RecastTime - (uint32)(time(0) - recast.TimeStamp))));
 
-        if (recast.ID != 0)
+        if (recast.ID == 256) // borrowing this id for mount recast
```

Comment:
what do you mean by borrowing?  if this ID (256) is written to 0xFE, doesn't that mean that it is actually 256?

Reply:
Well theres 8 bytes for each ability and the 4th byte is ability id.
They added 8 more bytes for mount recast and 4th byte is set to 0x01
Before calling a mount its just 0x00 and after recast expires the 0x01 hangs around
I just tested it without setting 0xFF to one and it still functions like retail.

Reply:
I figured I could shift the recast id one byte to make it match retail packet. If mount recast really is id 256 then it breaks the packets structure where one byte is used for recast id and 4 bytes of junk follow. Anyways, now that we know the client doesn't need that byte set, should I delete it?

Reply:
actually, going off what I'm seeing on windower's documentation on the packet, I think the abilityID is actually a uint16 and the recasttime is also a uint16.  Currently we're writing 4 bytes for the timestamp, and then overwriting the last byte with the ID.  But according to windower, the third byte of the timestamp is always zero (for abilities, of which.. none have a recast ID over 255).  So it seems very likely to me that a) mounts actually are recastid over 255 and b) both should write recasttime as a uint16 and then ability ID as another uint16

Reply:
That's all correct, but if mount is recast 256 then they're shifting it to start at the 3rd byte instead of the 4th byte to make room for an uint16, only for mounts, while the rest of the ability ids are stored in the 4th byte.

Reply:
Yeah there's no way ability ID is uint16, just possibly mount ID. I recall seeing the following 4 bytes full of junk.

Reply:
![ability_recasts](https://user-images.githubusercontent.com/7287342/37314736-f95a8478-2612-11e8-9908-996e57fb171e.png)

Reply:
Since ability recast IDs stop at 255, 256 was a safe value to borrow unless they add more abilities, and exceed the unused IDs.

Reply:
If they do that we have to fix this anyways, so I think its good can even remove the comment.

Reply:
yep, agree